internal/handlers/pg/pgdb/query.go

```diff
@@ -274,21 +299,16 @@ func prepareWhereClause(sqlFilters *types.Document) (string, []any) {
 			continue
 
 		case float64, string, types.ObjectID, int32, int64:
-			sql := fmt.Sprintf(
-				// Select if value under the key k is equal to value v.
-				`((_jsonb->%[1]s)::jsonb = %[2]s)`+
-					// If it's not, but the value under the key k is an array - select if it contains the value equal to v.
-					` OR (_jsonb->%[1]s)::jsonb @> %[2]s`,
-				p.Next(), // placeholder $1 used for field key k for preventing SQL injections
+			filters = append(filters, fmt.Sprintf(
+				// If the value under the key is not equal to v,
+				// but the value under the key k is an array - select if it contains the value equal to v.
+				sql+` OR (_jsonb%[1]s%[2]s)::jsonb @> %[3]s`,
+				operator, // -> or #>
+				p.Next(), // placeholder $1 used for field key or it's path for preventing SQL injections
```

Comment:
```suggestion
		// Select if value under the key is equal to provided value
		// If the value under the key is not equal to v,
		// but the value under the key k is an array - select if it contains the value equal to v.
		sql := `((_jsonb%[1]s%[2]s)::jsonb = %[3]s) OR (_jsonb%[1]s%[2]s)::jsonb @> %[3]s`
		
		// operator is -> for non-array, #> for array
		// placeholder p.Next() is used to prevent SQL injections.
		// The first p.Next() gets $1 and used for key,
		// the second p.Next() gets $2 and used for value.
		filters = append(filters, fmt.Sprintf(sql, operator,  p.Next(), p.Next())
```
Because you ask for suggestion :see_no_evil:. I know you've separated SQL to keep common element, it's much easier to find args %1 %2 %3 if it's right next to each other. 

I didn't test this. So it might not even compile :sweat_smile:

Reply:
So we want to stay with previous approach where we have predefined queries for both cases, right?

Reply:
Yes please, I prefer having both part of sql query together. It's just I don't have to scroll up and down to read :hugs: